sycl/test/on-device/config/select_device.cpp

```diff
@@ -173,9 +173,9 @@ static std::vector<DevDescT> getAllowListDesc(std::string allowList) {
 
 int main() {
   bool passed = false;
-  std::string sycl_be("PI_OPENCL");
-  if (getenv("SYCL_BE")) {
-    sycl_be = getenv("SYCL_BE");
+  std::string sycl_be("opencl");
+  if (getenv("SYCL_DEVICE_FILTER")) {
+    sycl_be = getenv("SYCL_DEVICE_FILTER");
```

Comment:
I'm not sure if this change matches "NFC" criteria. It removes validation of `SYCL_BE` and adds validation of `SYCL_DEVICE_FILTER`.
Should we keep testing `SYCL_BE` even though it's deprecated?

Reply:
My point is - we probably should keep testing deprecated features until they are removed.

Reply:
I restored on-device/basic_tests/diagnostics/device-check.cpp test.